File: java/05_031.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class UndirectedGraph {
        private Map<Object, List<Object>> graph;
        
        public UndirectedGraph() {
            this.graph = new HashMap<>();
        }
        
        public void addRelationship(Object node1, Object node2) {
            // Ensure both nodes exist in the graph
            if (!this.graph.containsKey(node1)) {
                this.graph.put(node1, new ArrayList<>());
            }
            if (!this.graph.containsKey(node2)) {
                this.graph.put(node2, new ArrayList<>());
            }
                
            // Add bidirectional relationship
            this.graph.get(node1).add(node2);
            this.graph.get(node2).add(node1);
        }
    }
}

File: data structures and algorithms/cs/05_031.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class UndirectedGraph
   {
      private Dictionary<object, List<object>> graph;

      public UndirectedGraph()
      {
         this.graph = new Dictionary<object, List<object>>();
      }

      public void AddRelationship(object node1, object node2)
      {
         // Ensure both nodes exist in the graph
         if (!this.graph.ContainsKey(node1))
         {
            this.graph[node1] = new List<object>();
         }
         if (!this.graph.ContainsKey(node2))
         {
            this.graph[node2] = new List<object>();
         }

         // Add bidirectional relationship
         this.graph[node1].Add(node2);
         this.graph[node2].Add(node1);
      }
   }
}
